xbmc/addons/AddonDatabase.h

```diff
@@ -73,13 +73,24 @@ class CAddonDatabase : public CDatabase
 
   bool Search(const std::string& search, ADDON::VECADDONS& items);
 
-  /*! \brief Disable an addon.
-   Sets a flag that this addon has been disabled.  If disabled, it is usually still available on disk.
-   \param addonID id of the addon to disable
-   \param disable whether to enable or disable.  Defaults to true (disable)
-   \return true on success, false on failure
-   \sa IsAddonDisabled, HasDisabledAddons */
-  bool DisableAddon(const std::string &addonID, bool disable = true);
+  /*!
+   * \brief Disable an addon.
+   * Sets a flag that this addon has been disabled.  If disabled, it is usually still available on
+   * disk.
+   * \param addonID id of the addon to disable
+   * \param disabledReason the reason why the addon is being disabled
+   * \return true on success, false on failure
+   * \sa IsAddonDisabled, HasDisabledAddons, EnableAddon
+   */
+  bool DisableAddon(const std::string& addonID, const ADDON::AddonDisabledReason& disabledReason);
```

Comment:
As `AddonDisabledReason` is an enum there is no benefit in passing it by ref. I would pass it by val.

Reply:
I left it as it indicates it should not change and if replaced by a class later the signature would not change. But I guess that superfluous currently.